corehq/motech/repeaters/migrations/0003_id_fields.py

```diff
@@ -22,11 +22,34 @@ class Migration(migrations.Migration):
             field=models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID'),
         ),
         migrations.RunSQL(
+            # Requires "repeaters_repeatrecord" to be empty
             sql="""
+            CREATE FUNCTION set_default_repeaters_repeater_id()
+                RETURNS trigger LANGUAGE plpgsql AS $BODY$
+                BEGIN
+                    IF NEW.id_ IS NULL THEN
+                        NEW.id_ = NEW.repeater_id::uuid;
+                    END IF;
+                    RETURN NEW;
+                END
+                $BODY$;
+            SET CONSTRAINTS "repeaters_repeatreco_repeater_id_01b51f9d_fk_repeaters" IMMEDIATE;
+            ALTER TABLE "repeaters_repeatrecord"
+                DROP CONSTRAINT "repeaters_repeatreco_repeater_id_01b51f9d_fk_repeaters",
+                ADD COLUMN "repeater_id_" uuid,
+                ALTER COLUMN repeater_id DROP NOT NULL;
+            ALTER TABLE "repeaters_repeater" ADD COLUMN "id_" uuid;
+            ALTER TABLE "repeaters_repeater"
+                ADD CONSTRAINT "repeaters_repeater_id_key" UNIQUE ("id"),
+                DROP CONSTRAINT "repeaters_repeater_pkey",
+                ALTER COLUMN "id_" TYPE uuid USING "repeater_id"::uuid,
+                ADD CONSTRAINT "repeaters_repeater_pkey" PRIMARY KEY ("id_"),
+                ADD CONSTRAINT id_eq CHECK ("id_" = "repeater_id"::uuid);
+            CREATE TRIGGER repeaters_repeater_default_id BEFORE INSERT ON repeaters_repeater
+                FOR EACH ROW EXECUTE FUNCTION set_default_repeaters_repeater_id();
```

Comment:
Is this a pattern we've used elsewhere? I guess we don't have any nice `AutoUUIDField` that encapsulates this. Do other models that want something similar just rely on the UUID being created in the application code and passed in?

Reply:
> Is this a pattern we've used elsewhere?

Not that I know of, but it's the simplest solution I could think of to keep the old `repeater_id` field and the new `id_` field in sync while keeping the table schema backward compatible for processes running a code version prior to this change set.

> I guess we don't have any nice AutoUUIDField that encapsulates this. Do other models that want something similar just rely on the UUID being created in the application code and passed in?

The conventional way to do this is something like `UUIDField(default=uuid.uuid4)`, where Django assigns the default value on object instantiation. But that doesn't work for code that is unaware of the field, so it would require downtime to deploy.